server/src/endpoints/orderNewDrink.ts

```diff
@@ -0,0 +1,57 @@
+import { AuthenticatedEndpointFunction, LogFn, Message } from '../endpoint'
+import { minimizeUser, updateUserProfile, User } from '../user'
+import generators from '../generators'
+
+const orderNewDrink: AuthenticatedEndpointFunction = async (user: User, inputs: any, log: LogFn) => {
+  let thisDrinkSkeleton: string
+
+  let privateActionString: string
+
+  const oldItem = user.item
+
+  const drinkVessels = generators.drinkVessels
+  const drinkContents = generators.drinkContents
+  const drinkNames = generators.drinkNames
+  const drinkSkeletons = generators.drinkSkeletons
+
+  const thisDrinkVessel = drinkVessels.generate()
+  const thisDrinkContent = drinkContents.generate()
+  const thisDrinkName = drinkNames.generate()
+  /* These three variables need to be put into the database to be retrieved when someone orders an existing drink. */
+  thisDrinkSkeleton = drinkSkeletons.generate()
+
+  thisDrinkSkeleton = thisDrinkSkeleton.replace('-drinkVessel-', thisDrinkVessel)
+  thisDrinkSkeleton = thisDrinkSkeleton.replace('-drinkContent-', thisDrinkContent)
+  thisDrinkSkeleton = thisDrinkSkeleton.replace('-drinkName-', thisDrinkName)
+  thisDrinkSkeleton = thisDrinkSkeleton.replace('-userName-', user.realName)
+
+  const item = thisDrinkName
+
+  const actionString = (item ? `picks up ${item}.` : `drops ${oldItem}.`)
```

Comment:
If someone doesn't have an item, this will generate a "so-and-so picks up [drink]" message, but if they do have an item, it will just say "so-and-so drops [old item]", with no reference to the new item they have. Just want to confirm that's the behavior you want instead of the "drops" message to also reference the new item.

Reply:
Oooh ok I will fix that right now!

Reply:
I'm actually not sure how to make that happen? I'm using the code from pickUpItem for this.

Reply:
Still trying to work this out, I don't think I understand the syntax. I would like it to do both, as with the normal pickUpItem.

Reply:
Ohhh, I'm being silly and misread this. The behavior here is "if the user is not trying to pick up an item (read: they clicked the "drop item" button), tell them they dropped their old item. Otherwise, quietly drop the old item and display they grabbed the new one"

Reply:
Ooooh, ok, that makes sense.

Reply:
In case that syntax is confusing, it's called a "ternary". A statement that looks like `x ? y : z` is roughly equivalent to `if(x) { y } else { z }`.

(In JS, if you try to take something that isn't a boolean, like the `item` var here, and ask if it's `true`, the rules are generally "does this thing exist / is it defined? if so, it's considered `true`", with some weird unexpected exceptions like the number 0 is `false` and the string `""` is false)